src/main/java/uk/gov/register/presentation/view/EntryListView.java

```diff
@@ -5,34 +5,43 @@
 import uk.gov.register.presentation.config.PublicBody;
 import uk.gov.register.presentation.dao.Entry;
 import uk.gov.register.presentation.representations.CsvRepresentation;
+import uk.gov.register.presentation.resource.NewPagination;
 import uk.gov.register.presentation.resource.Pagination;
 import uk.gov.register.presentation.resource.RequestContext;
 
+import java.util.Collection;
 import java.util.List;
 import java.util.Optional;
 
 public class EntryListView extends CsvRepresentationView {
+    private NewPagination newPagination;
     private Pagination pagination;
-    private List<Entry> entries;
+    private Collection<Entry> entries;
 
     public EntryListView(RequestContext requestContext, Pagination pagination, PublicBody custodian, Optional<GovukOrganisation.Details> custodianBranding, List<Entry> entries) {
         super(requestContext, custodian, custodianBranding, "entries.html");
         this.pagination = pagination;
         this.entries = entries;
     }
 
+    public EntryListView(RequestContext requestContext, NewPagination newPagination, PublicBody custodian, Optional<GovukOrganisation.Details> custodianBranding, Collection<Entry> entries, String template) {
+        super(requestContext, custodian, custodianBranding, template);
+        this.newPagination = newPagination;
+        this.entries = entries;
+    }
+
     @JsonValue
-    public List<Entry> getEntries() {
+    public Collection<Entry> getEntries() {
         return entries;
     }
 
     @SuppressWarnings("unused, used from templates")
-    public Pagination getPagination() {
-        return pagination;
+    public NewPagination getPagination() {
+        return newPagination;
     }
 
```

Comment:
This change breaks the /record/xxxx/entries resource. If you go to that resource it produces a template error because the newPagination property is null (the pagination property is set instead for this case and never returned to the view).

Suggest that EntryListView is changed to use single pagination property and constructor, either:

* Pagination and NewPagination classes should implement a single interface called IPagination, for example. Then getPagination() should return this and there should be one constructor that takes an IPagination.
* or it should only take a NewPagination property for both resources and the getAllEntriesOfARecord() method in the recordResource should create a hardcoded NewPagination object similarly to how it does currently with a Pagination object.